Fix compile error in  Volumetric test for gcc-14

lvk88 opened the pull request:
Currently it is not possible to build lib3mf master on Debian Trixie with gcc 14. This is because inclusion of `<algorithm>` is needed in order to use std::find, but it was missing, leading to the compiler error.

This pull request fixes this issue.

vijaiaeroastro commented:
~@lvk88 We do not accept PR's for master. We only do development on develop branch which gets merged into master after a release. This will be fixed in the next release as it is already fixed in develop.~

I see that you have made the PR for an example instead. But your message is confusing as you have copy pasted the message from your previous PR which was originally meant for master. Could you clarify here that you are actually trying to update an unit test in develop ? If so, perhaps this affects other examples / unit tests too.

Also, would this really affect the Debian build as you don't need unit tests to build lib3mf itself and use in Debian ?

lvk88 committed:
Fix compile errors for gcc-14

Inclusion of <algorithm> is needed in order to use std::find, but it was
missing, leading to compiler errors with gcc-14 on Debian Trixie.

lvk88 commented:
Hey @vijaiaeroastro , thanks for the quick feedback. I am sorry to have confused you. 

> But your message is confusing as you have copy pasted the message from your previous PR which was originally meant for master. 

I copy pasted the message from #420 because this PR achieves the same thing: it includes `<algorithm>` in a .cpp file where `std::find` is used. On older compilers, it is not a problem, but on g++14, it is a problem.

> Could you clarify here that you are actually trying to update an unit test in develop 

I am trying to fix a compiler error in a unit test on the develop branch. The unit test itself is not changed by this PR.

> If so, perhaps this affects other examples / unit tests too.

I don't think it affects other examples. After the fix, I can build the entire lib3mf, including unit tests.

> Also, would this really affect the Debian build as you don't need unit tests to build lib3mf itself and use in Debian ?

I want to be able to build lib3mf including unit tests on a Debian system with g++14, which is the default compiler on Debian Trixie. So in this sense, I guess the answer to your question is _yes, it does affect the Debian build_. Note that the same compiler error happens on Ubuntu 24.04 and Fedora 42, if you use g++-14. I just tried these other two distros.

I hope this clarifies all your questions. If not, please let me know.

vijaiaeroastro reviewed (approved):
Looks good.
